Sources/OpenAPIRuntime/URICoder/Parsing/URIParser.swift

```diff
@@ -302,17 +349,55 @@ extension String.SubSequence {
         return finalize(.foundSecondOrEnd)
     }
 
+    
     /// Accumulates characters until the provided character is found,
     /// or the end is reached. Moves the underlying startIndex.
-    /// - Parameter character: A character to stop at.
+    /// - Parameters:
+    ///   - startingCharacter: A character to start with.
+    ///   - endingCharacter: A character to stop at.
+    ///   If not provided or not found then uses the current start index as a starting character.
     /// - Returns: The accumulated substring.
-    fileprivate mutating func parseUpToCharacterOrEnd(_ character: Character) -> Self {
-        let startIndex = startIndex
+    fileprivate mutating func parseUpToCharacterOrEnd(startingCharacter: Character? = nil, _ endingCharacter: Character) -> Self {
```

Comment:
Can we take one more look here? Could you try to revert this part and go back to the original parseUpToCharacterOrEnd(_ character: Character) function, and explain why it can't work for you?

Reply:
Certainly. I need a method that can extract the nested key "bar" from the key: foo[bar]. The original method doesn't allow me to retrieve this nested key. When I pass "[" as a parameter, I receive "foo", and when I pass "]" I get "foo[bar". Am I missing something here?

Reply:
Right, that needs to happen in two steps:
1. Pass `[`, and get `foo` (the top level key)
2. Increment the iterator to the next character, because you know you are currently pointed at `[`.
3. Pass `]`, and get `bar` (the nested key)
4. Increment the iterator to move beyond the `]`.

Reply:
Ok now I got it. Much better this way. Thank you :)